dataactcore/models/jobTrackerInterface.py

```diff
@@ -285,3 +285,63 @@ def getSubmissionStatus(self,submissionId,interfaces):
             else:
                 return "validation_successful"
         return "unknown"
+
+    def getSubmissionById(self, submissionId):
+        """ Return submission object that matches ID"""
+        query = self.session.query(Submission).filter(Submission.submission_id == submissionId)
+        return self.runUniqueQuery(query, "No submission with that ID", "Multiple submissions with that ID")
+
+    def getSubmissionErrorInfo(self, submissionId):
+        """ Set number of errors and warnings for submission """
+        submission = self.getSubmissionById(submissionId)
+        submission.number_of_errors = self.interfaces.errorDb.sumNumberOfErrorsForJobList(self.getJobsBySubmission(submissionId), self.interfaces.validationDb)
+        submission.number_of_warnings = self.interfaces.errorDb.sumNumberOfErrorsForJobList(self.getJobsBySubmission(submissionId), self.interfaces.validationDb, errorType = "warning")
+
+    def setJobNumberOfErrors(self, jobId, numberOfErrors, errorType):
+        """ Label nuber of errors or warnings for specified job
+
+        Args:
+            jobId: Job to set number for
+            numberOfErrors: Number to be set
+            errorType: Type of error to set, can be either 'fatal' or 'warning'
+
+        """
+        job = self.getJobById(jobId)
+        if errorType == "fatal":
+            job.number_of_errors = numberOfErrors
+        elif errorType == "warning":
+            job.number_of_warnings = numberOfErrors
+
+    def setPublishableFlag(self, submissionId, publishable):
+        """ Set publishable flag to specified value """
+        submission = self.getSubmissionById(submissionId)
+        submission.publishable = publishable
+        self.session.commit()
+
+    def setPublishStatus(self, statusName, submissionId = None, submission = None):
+        """ Set publish status to specified name"""
+        statusId = self.getPublishStatusId(statusName)
+        if submission is None:
+            if submissionId is None:
+                raise ValueError("Must call setPublishStatus with either submission or submission ID")
+            submission = self.getSubmissionById(submissionId)
```

Comment:
Minor suggestion for aiding static analysis tools:
```python
def setPublishStatus(self, statusName, submissionOrId):
    if isinstance(submissionOrId, str):
        submission = self.getSubmissionById(submissionOrId)
    else:
        submission = submissionOrId
```
Same number of lines but errors can be found by linters rather than waiting for the runtime.